keras/integration_test/distributed_training_test.py

```diff
@@ -73,11 +76,40 @@ def dataset_fn(input_context):
         with strategy.scope():
             model = tf.keras.Sequential([tf.keras.layers.Dense(10)])
             optimizer = tf.keras.optimizers.SGD()
-            model.compile(optimizer, loss="mse", steps_per_execution=10)
+            model.compile(optimizer, loss="mse", steps_per_execution=5)
 
         x = tf.keras.utils.experimental.DatasetCreator(dataset_fn)
 
-        model.fit(x, epochs=2, steps_per_epoch=10)
+        logdir = os.path.join(self.get_temp_dir(), "logdir")
+        model.fit(
+            x,
+            epochs=2,
+            steps_per_epoch=20,
+            callbacks=[
+                tf.keras.callbacks.TensorBoard(
+                    logdir,
+                    update_freq=5,
+                    write_steps_per_second=True,
+                )
+            ],
+        )
+
+        events = []
+        for event_file in glob.glob(logdir + "/train/events.out.*"):
+            for event in tf.compat.v1.train.summary_iterator(event_file):
+                if not event.summary:
+                    continue
+                for value in event.summary.value:
+                    if value.tag != "batch_loss":
+                        continue
+                    events += [event.step]
+        events.sort()
+
+        if not isinstance(
+            strategy, tf.distribute.experimental.ParameterServerStrategy
```

Comment:
Can we also assert the case for PSS (which I assume has an empty `events`) and add a comment for why?